include/core/ib_particle.h

```diff
@@ -85,47 +85,69 @@ class IBParticle
 
   // This class defines values related to a particle used in the sharp interface
   // IB. Each particle defined will have these value used in the solver.
-  Point<dim>     position;
-  Point<dim>     last_position;
-  Tensor<1, dim> forces;
-  Tensor<1, dim> last_forces;
-  unsigned int   particle_id;
+  Point<dim>              position;
+  std::vector<Point<dim>> last_position;
+  Tensor<1, dim>          forces;
+  Tensor<1, dim>          last_forces;
+  unsigned int            particle_id;
 
   Tensor<1, 3> torques;
+  Tensor<1, 3> last_torques;
   double       mass;
   Tensor<2, 3> inertia;
   // Translational velocity
   Tensor<1, dim> velocity;
   // Store the last velocity of the fix point iteration.
-  Tensor<1, dim> last_velocity;
-  Tensor<1, dim> velocity_iter;
+  std::vector<Tensor<1, dim>> last_velocity;
```

Comment:
What is last velocity? if it is the previous, use previous instead :).
Last means the last one... you know ;)